Framework/Headers/OVCINDataTable.h

```diff
@@ -163,6 +182,60 @@ namespace OpenVanilla {
                 *iter = tolower(*iter);
         }
         
+        void detectMixedCase()
+        {
+            // detectMixedCase(): make m_isPairMapMixedCase = true if the pair map has mixed case
+            bool lowerCaseExist = false;
+            bool upperCaseExist = false;
+            
+            for (size_t index = 0; index < m_size; index++) {
+                const pair<string, string> entry = keyValuePairAtIndex(index);
+                string key = entry.first;
+                
+                if (!lowerCaseExist || !upperCaseExist) {
+                    for (int i = 0; i < key.length(); i++) {
+                        if (isupper(key.at(i))) {
+                            upperCaseExist = true;
+                            break;
+                        } else if (islower(key.at(i))) {
+                            lowerCaseExist = true;
+                            break;
+                        }
+                    }
+                } else if (lowerCaseExist && upperCaseExist) {
+                    m_isPairMapMixedCase = true;
+                    return;
+                }
+            }
+        }
+        
+        void insensitivizeMap(){
+            if (m_isPairMapMixedCase)
+                return;
+            
+            // if the map is not case mixed, make all the alphabet lowercase
+            for (size_t index = 0; index < m_size; index++) {
+                const pair<string, string> entry = keyValuePairAtIndex(index);
```

Comment:
醬改會得到兩個 err message.
1. Expected ')'
2. Use of undeclared identifier 'c'.
請問是哪邊有出問題呢？

Reply:
不好意思，沒有自己 debug 就貼了 code review comment。應該這樣改才對：

```
// Use m_index since the map may not have been frozen yet.
for (size_t index = 0; index < m_index; index++) {
    fprintf(stderr, "### %lu\n", index);
    
    KVPair* entry = m_data + index;
    char* key = entry->key;
    while (char c = *key) {
        if (isupper(c)) *key = tolower(c);
        ++key;
    }
}
```

請試試看這樣是否能正確讀入你需要的表格。

Reply:
感謝指點！初步測試沒有問題。